reactive/kubernetes_worker.py

```diff
@@ -242,6 +242,49 @@ def shutdown():
     service_stop('snap.kube-proxy.daemon')
 
 
+@when_not('config.changed.open-ports')
+def open_close_ports_if_needed():
+
+    # 1) Check which ports are open and save to already_opened
+    # opened_ports return format: "icmp" or "80/tcp"
+    already_opened = hookenv.opened_ports()
+    to_open = set(p.strip() for p in hookenv.config().get("open-ports","").split(','))
+    # first need to check if to_open has ports on right format, which are:
+    # PORT --> PORT is a number or icmp
+    # PORT/PROTOCOL --> PROTOCOL can only be tcp or udp
+    # PORT_1-PORT_2 --> both port codes must be numbers
+    # PORT_1-PORT_2/PROTOCOL --> both port codes must be numbers and protocol tcp or udp
+    for p in to_open:
+        port = p
+        if "/" in p:
+            port = p.split("/")[0]
+            if p.split("/")[1] != "tcp" and p.split("/")[1] != "udp":
+                # Log is for unit test purposes
+                hookenv.log("[ERROR] open-ports have wrong port format on: {}. Only tcp or udp allowed as protocols on PORT/PROTOCOL".format(p))
+                raise Exception("[ERROR] open-ports have wrong port format on: {}. Only tcp or udp allowed as protocols on PORT/PROTOCOL".format(p))
+        if not "-" in port:
+            if port != "icmp" and not port.isdigit():
```

Comment:
port is from to_open, but to_open's config says the valid values are tcp or udp only. There is a comment saying that icmp can be returned from hookenv.opened_ports, but that isn't this variable's source. Is this just extra that doesn't need to be here?